codedigger/atcoder/utils.py

```diff
@@ -0,0 +1,55 @@
+import re
+import json
+import requests
+from bs4 import BeautifulSoup as bs4
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from .scrapers import get_user_history
+from .api import get_user_results
+
+from lists.utils import get_next_url, get_prev_url, get_total_page
+from user.exception import ValidationException
+from codeforces.api import user_status
+
+
+def atcoder_status(handle):
+
+    res = get_user_history(handle)
+
+    contests_details = set()
+    all_contest = set()
+    solved = set()
+    wrong = set()
+
+    if res.status_code != 200:
+        return (contests_details, all_contest, solved, wrong)
+
+    soup = bs4(res.content, 'html5lib')
```

Comment:
Move all this content in a separate function in separate file say scraper_utils. This function will get user history and return the contests list.